cmd/build.go

```diff
@@ -82,6 +91,15 @@ it will be built and pushed`,
 		ref, _ := cmdFlags.GetString(flagGitRef)
 		tag, _ := cmdFlags.GetString(flagTag)
 		org, _ := cmdFlags.GetString(flagOrg)
+		repo, _ := cmdFlags.GetString(flagRepo)
+		repoHost, _ := cmdFlags.GetString(flagRepoHost)
+		dockerfile, _ := cmdFlags.GetString(flagDockerfile)
+		buildDir, _ := cmdFlags.GetString(flagBuildDir)
+		preBuild, _ := cmdFlags.GetString(flagPreBuild)
+		buildTarget, _ := cmdFlags.GetString(flagBuildTarget)
+		buildEnv, _ := cmdFlags.GetString(flagBuildEnv)
+		binaries, _ := cmdFlags.GetString(flagBinaries)
+		libraries, _ := cmdFlags.GetString(flagLibraries)
```

Comment:
Instead of a whole bunch of variables, an alternative pattern is using a single struct with many fields. 

E.g.

```go
type configFlags struct {
    Org string
    Repo string
    // ... etc.
}
```

Then:

```go
var config configFlags // Probably at the package level

// Set the field on the struct pointer. 
buildCmd.PersistentFlags().StringVar(&config.Repo, "repo", "", "Git repo override for building from a fork")
```

And now you can pass a single struct around vs. a bunch of variables. You also wouldn't need to use the viper `GetString()` and other methods. You can instead access the field on the struct.

Reply:
thanks! this cleaned it up significantly